### PR TITLE
Extend scenario metadata

### PR DESCRIPTION
**Pull request by Gamenot**
This PR adds additional information to the metadata. This allows for constrained and unconstrained arguments to be included in the metadata.

I am considering adding additional serialization support.

### CODE REVIEW

File: smarts/sstudio/types/scenario.py [hunk 1]
```diff
@@ -34,15 +39,98 @@
...
+    actor_of_interest_color = enum.auto()
+    actor_of_interest_re_filter = enum.auto()
+    scenario_difficulty = enum.auto()
+    scenario_length = enum.auto()
+
+
 @dataclass(frozen=True)
 class ScenarioMetadata:
-    """Scenario data that does not have influence on simulation."""
+    """Deprecated. Scenario data that does not have influence on simulation."""
```
Comment (saulfield) [inline smarts/sstudio/types/scenario.py:51]: Is there a reason we didn't just change this class? I like the original name better.

File: smarts/sstudio/types/scenario.py [hunk 2]
```diff
@@ -34,15 +39,98 @@
...
 
     actor_of_interest_re_filter: str
     """Vehicles with names that match this pattern are vehicles of interest."""
     actor_of_interest_color: Colors
     """The color that the vehicles of interest should have."""
 
+    def __post_init__(self):
+        warnings.warn(f"Use of {ScenarioMetadata.__name__} is deprecated.")
```
Comment (saulfield) [inline smarts/sstudio/types/scenario.py:59]: Might want to suggest the new class to use instead.

File: smarts/sstudio/types/scenario.py [hunk 3]
```diff
@@ -34,15 +39,98 @@
...
+
+
+class StandardMetadata(collections.Mapping):
+    """Metadata that does not have direct on simulation."""
```
Comment (saulfield) [inline smarts/sstudio/types/scenario.py:63]:
```suggestion
    """Metadata that does not have direct influence on the simulation."""
```

File: smarts/sstudio/types/scenario.py [hunk 4]
```diff
@@ -34,15 +39,98 @@
...
+
+        Returns:
+            Optional[Any]: The value or default.
+        """
+        if isinstance(__key, MetadataFields):
+            __key = __key.name
+        return self._dict_metadata.get(__key, __default)
+
+    # def __getattr__(self, __name: str) -> Any:
+    #     return self._dict_metadata.get(__name)
```
Comment (saulfield) [inline smarts/sstudio/types/scenario.py:121]: Remove or uncomment?

File: scenarios/sumo/platoon/merge_exit_sumo_t_agents_1/scenario.py [hunk 5]
```diff
@@ -10,15 +10,14 @@
...
-    Mission,
     Route,
     Scenario,
-    ScenarioMetadata,
     Traffic,
     TrafficActor,
     TrapEntryTactic,
     Trip,
 )
+from smarts.sstudio.types.scenario import StandardMetadata
```
Comment (saulfield) [inline scenarios/sumo/platoon/merge_exit_sumo_t_agents_1/scenario.py:20]: Minor, but makes sense to import it from `smarts.sstudio.types` above with the other imports.

File: smarts/sstudio/types/scenario.py (hunk 1 above)
Comment (Gamenot) [reply smarts/sstudio/types/scenario.py]: Right, I had changed this already by the time you commented. I forgot to push the updates here.

**Review by saulfield — APPROVED** [review]

**Comment by Gamenot** [comment]
Something odd is happening with the `_build` directory now. I wonder if it has to do with using `yaml` instead of `pickle`.

**Review by qianyi-sun — APPROVED** [review]

**Pull request merged**